packages/cursorless-engine/src/ScopeVisualizer/ScopeVisualizerImpl.ts

```diff
@@ -0,0 +1,56 @@
+import { ScopeRenderer, showError } from "@cursorless/common";
+import { ModifierStageFactory } from "../processTargets/ModifierStageFactory";
+import { ScopeHandlerFactory } from "../processTargets/modifiers/scopeHandlers/ScopeHandlerFactory";
+import { ide } from "../singletons/ide.singleton";
+import { PerEditor } from "../util/PerEditor";
+import { EditorScopeVisualizer } from "./EditorScopeVisualizer";
+import { ScopeVisualizer } from "..";
+
+export class ScopeVisualizerImpl implements ScopeVisualizer {
```

Comment:
call this `ScopeRangeProvider`?  Then probably we extract the scope extraction code and potentially use it for #1524